books/RayTracingTheNextWeek.html

```diff
@@ -190,8 +190,8 @@
         public:
             moving_sphere() {}
             moving_sphere(
-                point3 cen0, point3 cen1, double t0, double t1, double r, shared_ptr<material> m)
-                : center0(cen0), center1(cen1), time0(t0), time1(t1), radius(r), mat_ptr(m)
+                point3 ctr0, point3 ctr1, double t0, double t1, double r, shared_ptr<material> m)
+                : center0(ctr0), center1(ctr1), time0(t0), time1(t1), radius(r), mat_ptr(m)
```

Comment:
Hmmmmm....

I don't want the member variables to be `center0`, `center1`, `time0`, `time1`.
We're already removing the `0` and `1` suffix from other places in the code.
However, `center_start`, `center_end`, `time_start`, `time_end` are potentially a _BAD_ idea, because someone might be confused into thinking that `time_start` and `time_end` are the same values as the camera's `time_start` and `time_end` when they very much might not be. The same goes for `center_start` and `center_end` being confused as being center location at time of `time_start` and `time_end`.

The sphere also just linearly interpolates off to infinity in both directions so the member variables are not `_start` and `_end` per se. I guess I'm fine leaving them as `*0` and `*1`, but I wonder if we'd be better off something else.

Wikipedia seems to think suffix with `*0` and `*1` are acceptable:
https://en.wikipedia.org/wiki/Linear_interpolation

Reply:
Seems unavoidable. Could be `time_start_center` and `time_end_center`, or `time0_center` and `time1_center`. Wordy, but definitely clear.